src/Core/Domain/Product/Combination/QueryHandler/GetCombinationIdsHandler.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Query\GetCombinationIds;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Grid\Query\ProductCombinationQueryBuilder;
+use PrestaShop\PrestaShop\Core\Search\Filters\ProductCombinationFilters;
+
+class GetCombinationIdsHandler implements GetCombinationIdsHandlerInterface
+{
+    /**
+     * @var ProductCombinationQueryBuilder
+     */
+    private $productCombinationQueryBuilder;
+
+    /**
+     * @param ProductCombinationQueryBuilder $productCombinationQueryBuilder
+     */
+    public function __construct(
+        ProductCombinationQueryBuilder $productCombinationQueryBuilder
+    ) {
+        $this->productCombinationQueryBuilder = $productCombinationQueryBuilder;
+    }
+
+    /**
+     * @param GetCombinationIds $query
+     *
+     * @return CombinationId[]
+     */
+    public function handle(GetCombinationIds $query): array
+    {
+        $filters = $query->getFilters();
+        $filters['product_id'] = $query->getProductId()->getValue();
+
+        $searchCriteria = new ProductCombinationFilters(
+            $query->getShopConstraint(),
+            [
+                'limit' => $query->getLimit(),
+                'offset' => $query->getOffset(),
+                'orderBy' => $query->getOrderBy(),
+                'sortOrder' => $query->getOrderWay(),
+                'filters' => $filters,
+            ]
+        );
+
+        $results = $this->productCombinationQueryBuilder
+            ->getSearchQueryBuilder($searchCriteria)
```

Comment:
```suggestion
            ->getSearchQueryBuilder($searchCriteria)
            ->select('pas.id_product_attribute')
```

You can override the select part so that the query only returns the required data

Reply:
yeah been there, but all selects are still needed as otherwise filters fails

Reply:
actually not all parts are needed, but only price, it fails to order by it and throws "nonUniqueFieldNameException" when ordering by price. 
So I think its enough if I select id_product attribute and price here

Reply:
this is what I ended up with, I think its fair
[8cfbb28](https://github.com/PrestaShop/PrestaShop/pull/30585/commits/8cfbb28eb5be626674ae9b78fb3666dc60b64b36)

Reply:
Yep, seems fine!